hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/util/Caches.java

```diff
@@ -223,10 +226,18 @@ public static AdvancedCache failSilentReadCache(AdvancedCache cache) {
 	 */
 	public static void broadcastEvictAll(AdvancedCache cache) {
 		final RpcManager rpcManager = cache.getRpcManager();
-		if ( rpcManager != null ) {
+		if (rpcManager != null) {
 			final EvictAllCommand cmd = new EvictAllCommand(ByteString.fromString(cache.getName()));
-			final RpcOptions options = rpcManager.getDefaultRpcOptions(isSynchronousCache(cache));
-			rpcManager.invokeRemotely(null, cmd, options);
+
+			final RpcOptions options;
+			if (isSynchronousCache(cache)) {
+				options = rpcManager.getSyncRpcOptions();
+			} else {
+				long timeout = cache.getCacheConfiguration().clustering().remoteTimeout();
+				options = new RpcOptions(DeliverOrder.PER_SENDER, timeout, TimeUnit.MILLISECONDS);
+			}
+
+			rpcManager.blocking(rpcManager.invokeCommandOnAll(cmd, VoidResponseCollector.validOnly(), options));
```

Comment:
How I did I miss this before!?
you shouldn't do that. `invokeCommandOnAll()` as it is always sync, i.e. it waits for a response.
you need to use `sendToAll()` for async.